src/coreclr/tools/Common/Compiler/NativeAotNameMangler.cs

```diff
@@ -433,17 +437,17 @@ static void AppendTypeName(Utf8StringBuilder sb, MetadataType t)
                     }
                     else if (type is IPrefixMangledMethod)
                     {
-                        mangledName = GetPrefixMangledMethodName((IPrefixMangledMethod)type).ToString();
+                        mangledName = GetPrefixMangledMethodName((IPrefixMangledMethod)type);
                     }
                     else if (type is IPrefixMangledType)
                     {
-                        mangledName = GetPrefixMangledTypeName((IPrefixMangledType)type).ToString();
+                        mangledName = GetPrefixMangledTypeName((IPrefixMangledType)type);
                     }
                     else
                     {
                         // This is a type definition. Since we didn't fall in the `is EcmaType` case above,
                         // it's likely a compiler-generated type.
-                        mangledName = SanitizeName(((DefType)type).GetFullName());
+                        mangledName = new Utf8String(SanitizeName(((DefType)type).GetFullName()));
```

Comment:
This has unnecessary conversions too. Can be something along the lines of:

```suggestion
                        var defType = (DefType)type;
                        var sb = new Utf8StringBuilder();
                        if (defType.Namespace.Length > 0)
                            sb.Append(SanitizeName(defType.Namespace)).Append('_');
                        sb.Append(SanitizeName(defType.Name));
                        mangledName = sb.ToString();
```